File: Java/Hash Table/1. Two Sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> memo = new HashMap<>();
        for (int i = 0, n = nums.length; i < n; i++){
            int num = nums[i];
            if (memo.containsKey(num)) return new int[]{memo.get(num), i};
            memo.put(target - num, i);
        }
        return new int[2];
    }
}

File: Kotlin/Hash Table/1. Two Sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var memo : HashMap<Int, Int> = HashMap<Int, Int> ()
        for (i in 0..nums.size-1){
            var num = nums[i]
            if (memo.containsKey(num)) {
                return intArrayOf(memo.get(num)!!.toInt(), i)
            }
            memo.put(target - num, i)
        }
        return intArrayOf(-1, -1)
    }
}